mainwindow.cpp

```diff
@@ -23,9 +23,9 @@ MainWindow::MainWindow(QWidget *parent)
 
 MainWindow::~MainWindow() { delete ui; }
 
-void MainWindow::on_increaseBattery_clicked() { battery.increase_battery(1); }
+void MainWindow::on_increaseBattery_clicked() { battery.increase_battery(1); battery.CheckRange(); }
```

Comment:
Can you check the ranges when the increase and decrease buttons are clicked, this way we only have to call the increase and decrease functions 
```suggestion
void MainWindow::on_increaseBattery_clicked() { battery.increase_battery(1); }
```